### PR TITLE
ILCompiler: don't strip debug info from 'ilc' during source-build.

### PR DESCRIPTION
**Pull request by tmds**
@jkotas ptal.

cc @omajid @ViktorHofer

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <!-- Disable native AOT on FreeBSD when cross building from Linux. -->
     <NativeAotSupported Condition="'$(TargetOS)' == 'freebsd' and '$(CrossBuild)' == 'true'">false</NativeAotSupported>
     <PublishAot Condition="'$(NativeAotSupported)' == 'true'">true</PublishAot>
+    <StripSymbols Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(KeepNativeSymbols)' == 'true'">false</StripSymbols>
```
Comment (tmds) [inline src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj:19]: `KeepNativeSymbols` is initialized as

https://github.com/dotnet/runtime/blob/ae6ad886bb37297912689a4edd4d29988585d29a/Directory.Build.props#L393-L394

We _could_ leave out the `DotNetBuildSourceOnly` condition here, if desired.

**Comment by MichalStrehovsky** [comment]
It might be better to add this to https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/aot/AotCompilerCommon.props. I expect we'd need the same for source-built crossgen2 eventually and this file sets shared policies.

This project file is getting overhauled in #103375 and if for nothing else, to avoid a conflict.

File: src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj (hunk 1 above)
Comment (tmds) [reply src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj]: @MichalStrehovsky do you want to keep this as is, or remove the `DotNetBuildSourceOnly` condition?

Comment (MichalStrehovsky) [reply src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj]: Dropping the `'$(DotNetBuildSourceOnly)' == 'true'` part sounds good to me.

**Review by jkotas — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**